org.scala-ide.sdt.core/src/org/scalaide/core/internal/project/ScalaProject.scala

```diff
@@ -427,20 +427,27 @@ class ScalaProject private (val underlying: IProject) extends ClasspathManagemen
   /** Compiler settings that are honored by the presentation compiler. */
   private def isPCSetting(settings: Settings): Set[Settings#Setting] = {
     import settings.{ plugin => pluginSetting, _ }
-    Set(deprecation,
+
+    val compilerPluginSettings: Set[Settings#Setting] = Set(pluginOptions,
+      pluginSetting,
+      pluginsDir)
+
+    val generalSettings: Set[Settings#Setting] = Set(deprecation,
       unchecked,
-      pluginOptions,
       verbose,
       Xexperimental,
       future,
       Ylogcp,
-      pluginSetting,
-      pluginsDir,
       YpresentationDebug,
       YpresentationVerbose,
       YpresentationLog,
       YpresentationReplay,
       YpresentationDelay)
+
+    if (effectiveScalaInstallation().version == ScalaInstallation.platformInstallation.version)
+      generalSettings ++ compilerPluginSettings
+    else
+      generalSettings
```

Comment:
I don't understand why this solves the problem. I have a test project, which is configured to use 2.11.6 paradise plugin. But my IDE has version 2.11.7. With this change the PC errors are gone (semantic highlighting on the other side doesn't work anymore) but where does the 2.11.7 paradise plugin come from? `pluginsDir` is set to `-Xpluginsdir = misc/scala-devel/plugins`, where does this directory point to?

Should we always ignore user plugin preferences for the PC? If the wrong version of a compiler plugin is used and the PC is configured with this one, we will get error markers whenever the compiler plugin is used.